File: crates/sui-python/examples/counter.py
import sui


count = sui.State(0)
enabled = sui.State(True)
opacity = sui.State(0.65)
name = sui.State("")


def increment():
    count.set(count.get() + 1)


theme = sui.Theme.dark()
app = sui.App(theme=theme)
root = sui.column(
    [
        sui.label(count),
        sui.button("Increment", on_press=increment),
        sui.checkbox("Enabled", enabled),
        sui.switch("Preview", True),
        sui.slider("Opacity", opacity, min_value=0.0, max_value=1.0, step=0.05),
        sui.text_input("Name", name, placeholder="Optional label"),
    ],
    gap=8,
)
app.window(sui.Window("Counter").root(root))

running = app.start()
print("initial commands:", running.render().command_count)

running.ui_handle().post(increment)
running.drain()
print("count:", count.get())
print("updated commands:", running.render().command_count)

theme.set_preset("light")
running.drain()
print("theme updated commands:", running.render().command_count)

File: crates/sui-js/examples/counter.js
"use strict";

const sui = require("..");

const count = new sui.State(0);
const enabled = new sui.State(true);
const opacity = new sui.State(0.65);
const name = new sui.State("");

function increment() {
  count.set(Number(count.get()) + 1);
}

const theme = sui.Theme.dark();
const root = sui.column(
  [
    sui.label(count),
    sui.button("Increment", { onPress: increment }),
    sui.checkbox("Enabled", { checked: enabled }),
    sui.switchControl("Preview", { on: true }),
    sui.slider("Opacity", { value: opacity, min: 0, max: 1, step: 0.05 }),
    sui.textInput("Name", { value: name, placeholder: "Optional label" }),
  ],
  { gap: 8 }
);

const window = new sui.Window("Counter");
window.root(root);

const app = new sui.App();
app.setTheme(theme);
app.window(window);

const running = app.start();
console.log("initial commands:", running.render().commandCount);

running.uiHandle().post(increment);
running.drain();
console.log("count:", count.get());
console.log("updated commands:", running.render().commandCount);

theme.setPreset("light");
running.drain();
console.log("theme updated commands:", running.render().commandCount);
